alternator/executor.cc

```diff
@@ -96,6 +96,44 @@ std::string make_jsonable::to_json() const {
     return rjson::print(_value);
 }
 
+
+static bool is_leaf(const rjson::value& value) {
+    return !value.IsObject() && !value.IsArray();
+}
+
+static future<> destroy_gently_nonleaf(rjson::value&& value_in);
+
+future<> destroy_gently(rjson::value&& value) {
+    if (is_leaf(value)) {
+        return make_ready_future<>();
+    } else {
+        return destroy_gently_nonleaf(std::move(value));
+    }
+}
+
+static
+future<>
+destroy_gently_nonleaf(rjson::value&& value_in) {
+    // value_in is an rvalue ref to force caller to move
+    auto value = std::move(value_in);
+
+    if (value.IsObject()) {
+        for (auto it = value.MemberBegin(); it != value.MemberEnd();) {
+            // destroy_gently() is not a coroutine to avoid allocation if it->value is a leaf
```

Comment:
nit: why is this interesting here?
would the code look differently if it were a coroutine? (pass value by lvalue reference?)